docs/concepts/artifacts.md

```diff
@@ -0,0 +1,228 @@
+# Artifacts
+
+Artifacts offer a means to publish and manage crucial data that is rendered in the Prefect UI, making it easy to see where and when your results were written. Published artifacts may be associated with a particular task run, flow run, or outside a flow run context. Artifacts provide a more advanced way to present information compared to typical logging practices-- including the ability to display reports, tables, markdown, and links to external data.
```

Comment:
```suggestion
Artifacts offer a means to publish and manage data that is rendered in the Prefect UI, making it easy to see where and when your results were written. Published artifacts may be associated with a particular task run, flow run, or outside a flow run context. Artifacts provide a more advanced way to present information compared to typical logging practices-- including the ability to display tables, markdown, and links to external data.
```

I don't think data has to be crucial to be included in an artifact so we can keep it simple and call it data.  

Do we want the connection to the concept of results here?  @billpalombi - maybe a question for you.